vscode_extension/src/aacExecutableWrapper.ts

```diff
@@ -0,0 +1,123 @@
+        if(commandName) {
+            getAacCommandArgs(commandName).then(async (commandArguments) => {
+                await getCommandArgUserInput(commandArguments);
+                execAacShellCommand(commandName, commandArguments).then(output => {
+                    outputChannel.appendLine(output);
+                    outputChannel.show();
+                });
+            });
+        }
+    });
+}
+
+async function getCommandArgUserInput(commandArguments: CommandArgument[]) {
+
+    let unansweredCommandArguments: CommandArgument[] = commandArguments.filter(argument => { return (argument.userResponse === undefined); } );
+
+    if (unansweredCommandArguments.length > 0) {
+        let argumentToPromptFor = unansweredCommandArguments[0];
+        const dialogBoxOptions: OpenDialogOptions = {
+            title: argumentToPromptFor.name,
+        };
+        const inputBoxOptions: InputBoxOptions = {
+            title: argumentToPromptFor.name,
+            prompt: argumentToPromptFor.description,
+        };
+
+        if (argumentToPromptFor.description.toLowerCase().includes("path")){
+            let fileUri: Uri[] | undefined = await window.showOpenDialog(inputBoxOptions);
+            if(fileUri)
+                {argumentToPromptFor.userResponse = fileUri[0]?.path;}
+        } else {
+            argumentToPromptFor.userResponse = await window.showInputBox(inputBoxOptions);
+        }
+    } else if(unansweredCommandArguments.length > 1) {
+        getCommandArgUserInput(commandArguments);
+    }
+}
+
+async function getAacCommandNames(): Promise<string[]> {
+    const aacHelpOutput = await execAacShellCommand("-h");
+    return parseTaskNamesFromHelpCommand(aacHelpOutput);
+}
+
+/**
+ * @returns list of Command arguments
+ */
+async function getAacCommandArgs(aacCommandName: string): Promise<CommandArgument[]> {
+    const aacHelpOutput = await execAacShellCommand(`${aacCommandName} -h`);
+    return parseTaskArgsFromHelpCommand(aacHelpOutput);
+}
+
+async function execAacShellCommand(command: string, commandArgs: CommandArgument[] = []): Promise<string> {
+
+    let result: string = "";
+
+    let commandArgsArray = ["aac", command, ...(commandArgs.map(argument => argument.userResponse))];
+    try {
+        const { stdout, stderr } = await execShell(commandArgsArray.join(" "), {});
+        const stringOutput = stderr.length > 0 ? stderr : stdout;
+        return stringOutput;
+
+    } catch (error: any) {
+        let errorMessage = error.stderr || error.stdout || "";
+
+        outputChannel.appendLine(`Failed to execute AaC command:\n${errorMessage}`);
+        outputChannel.show(true);
+        throw error;
+    }
+}
+
+function parseTaskNamesFromHelpCommand(aacHelpOutput: string): string[] {
+
+    const regExp = /{(.*)}/;
+    const commandNamesMatch = regExp.exec(aacHelpOutput);
+
+    let commandNames: string[] = [];
+    if (commandNamesMatch && commandNamesMatch.length >= 2) {
```

Comment:
I'd prefer if we could extract this into a function that would convey better the significance of `commandNames && commandNamesMatch.length >= 2` - perhaps, something like `if (foundCommandNames()) {`, or something.

Reply:
I don't think this is really worth a separate function; it's a non-null check then a length check.

Reply:
Yes, it's clear that it's a null check and a length check, but it's not obvious (until you go figure out that the regex match groups start at position 1 of the array) why you're doing the length check. When I first looked at this, I had to figure out what the relevance of the array having more than two elements was. Unless someone is very familiar with JavaScript and/or TypeScript I don't think this would be obvious to them, either.